Description: Remove authentik.enterprise module references from:
- authentik/admin/api/system.py
- authentik/blueprints/v1/importer.py
- authentik/core/api/devices.py
- authentik/core/api/groups.py
- authentik/core/api/object_types.py
- authentik/core/api/users.py
- authentik/events/api/events.py
- authentik/outposts/api/outposts.py
- authentik/providers/scim/models.py

.
Enterprise related source code has been removed from build. Cleaning up authentik.enterprise module references since it no longer exists.
Author: Matthew Ramirez <matthew.ramirez@chainguard.dev>
Forwarded: not-needed

diff --git a/authentik/admin/api/system.py b/authentik/admin/api/system.py
index 28b0b86f4..0da2192b7 100644
--- a/authentik/admin/api/system.py
+++ b/authentik/admin/api/system.py
@@ -18,7 +18,7 @@ from rest_framework.views import APIView
 
 from authentik import authentik_full_version
 from authentik.core.api.utils import PassiveSerializer
-from authentik.enterprise.license import LicenseKey
+#from authentik.enterprise.license import LicenseKey
 from authentik.lib.config import CONFIG
 from authentik.lib.utils.reflection import get_env
 from authentik.outposts.apps import MANAGED_OUTPOST
@@ -81,7 +81,8 @@ class SystemInfoSerializer(PassiveSerializer):
             "authentik_version": authentik_full_version(),
             "environment": get_env(),
             "openssl_fips_enabled": (
-                backend._fips_enabled if LicenseKey.get_total().status().is_valid else None
+                #backend._fips_enabled if LicenseKey.get_total().status().is_valid else None
+                None
             ),
             "openssl_version": OPENSSL_VERSION,
             "platform": platform.platform(),
diff --git a/authentik/blueprints/tests/test_managed_app_config.py b/authentik/blueprints/tests/test_managed_app_config.py
index 784811035..3f27c45b1 100644
--- a/authentik/blueprints/tests/test_managed_app_config.py
+++ b/authentik/blueprints/tests/test_managed_app_config.py
@@ -1,14 +1,14 @@
 from django.test import TestCase
 
 from authentik.blueprints.apps import ManagedAppConfig
-from authentik.enterprise.apps import EnterpriseConfig
+#from authentik.enterprise.apps import EnterpriseConfig
 from authentik.lib.utils.reflection import get_apps
 
 
 class TestManagedAppConfig(TestCase):
     def test_apps_use_managed_app_config(self):
         for app in get_apps():
-            if app.name.startswith("authentik.enterprise"):
-                self.assertIn(EnterpriseConfig, app.__class__.__bases__)
-            else:
-                self.assertIn(ManagedAppConfig, app.__class__.__bases__)
+            #if app.name.startswith("authentik.enterprise"):
+            #    self.assertIn(EnterpriseConfig, app.__class__.__bases__)
+            #else:
+            self.assertIn(ManagedAppConfig, app.__class__.__bases__)
diff --git a/authentik/blueprints/v1/importer.py b/authentik/blueprints/v1/importer.py
index 9301edf6f..2eb58238a 100644
--- a/authentik/blueprints/v1/importer.py
+++ b/authentik/blueprints/v1/importer.py
@@ -42,21 +42,21 @@ from authentik.core.models import (
     User,
     UserSourceConnection,
 )
-from authentik.enterprise.license import LicenseKey
-from authentik.enterprise.models import LicenseUsage
-from authentik.enterprise.providers.google_workspace.models import (
-    GoogleWorkspaceProviderGroup,
-    GoogleWorkspaceProviderUser,
-)
-from authentik.enterprise.providers.microsoft_entra.models import (
-    MicrosoftEntraProviderGroup,
-    MicrosoftEntraProviderUser,
-)
-from authentik.enterprise.providers.ssf.models import StreamEvent
-from authentik.enterprise.stages.authenticator_endpoint_gdtc.models import (
-    EndpointDevice,
-    EndpointDeviceConnection,
-)
+#from authentik.enterprise.license import LicenseKey
+#from authentik.enterprise.models import LicenseUsage
+#from authentik.enterprise.providers.google_workspace.models import (
+#    GoogleWorkspaceProviderGroup,
+#    GoogleWorkspaceProviderUser,
+#)
+#from authentik.enterprise.providers.microsoft_entra.models import (
+#    MicrosoftEntraProviderGroup,
+#    MicrosoftEntraProviderUser,
+#)
+#from authentik.enterprise.providers.ssf.models import StreamEvent
+#from authentik.enterprise.stages.authenticator_endpoint_gdtc.models import (
+#    EndpointDevice,
+#    EndpointDeviceConnection,
+#)
 from authentik.events.logs import LogEvent, capture_logs
 from authentik.events.utils import cleanse_dict
 from authentik.flows.models import FlowToken, Stage
@@ -118,7 +118,7 @@ def excluded_models() -> list[type[Model]]:
         # Classes which are only internally managed
         # FIXME: these shouldn't need to be explicitly listed, but rather based off of a mixin
         FlowToken,
-        LicenseUsage,
+        #LicenseUsage,
         SCIMProviderGroup,
         SCIMProviderUser,
         Tenant,
@@ -133,14 +133,14 @@ def excluded_models() -> list[type[Model]]:
         WebAuthnDeviceType,
         SCIMSourceUser,
         SCIMSourceGroup,
-        GoogleWorkspaceProviderUser,
-        GoogleWorkspaceProviderGroup,
-        MicrosoftEntraProviderUser,
-        MicrosoftEntraProviderGroup,
-        EndpointDevice,
-        EndpointDeviceConnection,
+        #GoogleWorkspaceProviderUser,
+        #GoogleWorkspaceProviderGroup,
+        #MicrosoftEntraProviderUser,
+        #MicrosoftEntraProviderGroup,
+        #EndpointDevice,
+        #EndpointDeviceConnection,
         DeviceToken,
-        StreamEvent,
+        #StreamEvent,
         UserConsent,
         SAMLSession,
         Message,
@@ -197,7 +197,8 @@ class Importer:
     def default_context(self):
         """Default context"""
         return {
-            "goauthentik.io/enterprise/licensed": LicenseKey.get_total().status().is_valid,
+            #"goauthentik.io/enterprise/licensed": LicenseKey.get_total().status().is_valid,
+            "goauthentik.io/enterprise/licensed": False,
             "goauthentik.io/rbac/models": rbac_models(),
         }
 
diff --git a/authentik/core/api/devices.py b/authentik/core/api/devices.py
index d6ceb932a..bf3abbab9 100644
--- a/authentik/core/api/devices.py
+++ b/authentik/core/api/devices.py
@@ -15,7 +15,7 @@ from rest_framework.viewsets import ViewSet
 
 from authentik.core.api.users import ParamUserSerializer
 from authentik.core.api.utils import MetaNameSerializer
-from authentik.enterprise.stages.authenticator_endpoint_gdtc.models import EndpointDevice
+#from authentik.enterprise.stages.authenticator_endpoint_gdtc.models import EndpointDevice
 from authentik.stages.authenticator import device_classes, devices_for_user
 from authentik.stages.authenticator.models import Device
 from authentik.stages.authenticator_webauthn.models import WebAuthnDevice
@@ -42,16 +42,16 @@ class DeviceSerializer(MetaNameSerializer):
         """Get extra description"""
         if isinstance(instance, WebAuthnDevice):
             return instance.device_type.description if instance.device_type else None
-        if isinstance(instance, EndpointDevice):
-            return instance.data.get("deviceSignals", {}).get("deviceModel")
+        #if isinstance(instance, EndpointDevice):
+        #    return instance.data.get("deviceSignals", {}).get("deviceModel")
         return None
 
     def get_external_id(self, instance: Device) -> str | None:
         """Get external Device ID"""
         if isinstance(instance, WebAuthnDevice):
             return instance.device_type.aaguid if instance.device_type else None
-        if isinstance(instance, EndpointDevice):
-            return instance.data.get("deviceSignals", {}).get("deviceModel")
+        #if isinstance(instance, EndpointDevice):
+        #    return instance.data.get("deviceSignals", {}).get("deviceModel")
         return None
 
 
diff --git a/authentik/core/api/groups.py b/authentik/core/api/groups.py
index 95b87a868..ca0ae8c52 100644
--- a/authentik/core/api/groups.py
+++ b/authentik/core/api/groups.py
@@ -231,14 +231,14 @@ class GroupViewSet(UsedByMixin, ModelViewSet):
     def get_ql_fields(self):
         from djangoql.schema import BoolField, StrField
 
-        from authentik.enterprise.search.fields import (
-            JSONSearchField,
-        )
+        #from authentik.enterprise.search.fields import (
+        #    JSONSearchField,
+        #)
 
         return [
             StrField(Group, "name"),
             BoolField(Group, "is_superuser", nullable=True),
-            JSONSearchField(Group, "attributes", suggest_nested=False),
+            #JSONSearchField(Group, "attributes", suggest_nested=False),
         ]
 
     def get_queryset(self):
diff --git a/authentik/core/api/object_types.py b/authentik/core/api/object_types.py
index 12acd4e4a..7997864d9 100644
--- a/authentik/core/api/object_types.py
+++ b/authentik/core/api/object_types.py
@@ -10,7 +10,7 @@ from rest_framework.request import Request
 from rest_framework.response import Response
 
 from authentik.core.api.utils import PassiveSerializer
-from authentik.enterprise.apps import EnterpriseConfig
+#from authentik.enterprise.apps import EnterpriseConfig
 from authentik.lib.utils.reflection import all_subclasses
 
 
@@ -66,9 +66,9 @@ class TypesMixin:
                         "component": instance.component,
                         "model_name": subclass._meta.model_name,
                         "icon_url": getattr(instance, "icon_url", None),
-                        "requires_enterprise": isinstance(
-                            subclass._meta.app_config, EnterpriseConfig
-                        ),
+                        #"requires_enterprise": isinstance(
+                        #    subclass._meta.app_config, EnterpriseConfig
+                        #),
                     }
                 )
             except NotImplementedError:
diff --git a/authentik/core/api/users.py b/authentik/core/api/users.py
index 2544f5efe..8cd69631d 100644
--- a/authentik/core/api/users.py
+++ b/authentik/core/api/users.py
@@ -436,10 +436,10 @@ class UserViewSet(UsedByMixin, ModelViewSet):
     def get_ql_fields(self):
         from djangoql.schema import BoolField, StrField
 
-        from authentik.enterprise.search.fields import (
-            ChoiceSearchField,
-            JSONSearchField,
-        )
+        #from authentik.enterprise.search.fields import (
+        #    ChoiceSearchField,
+        #    JSONSearchField,
+        #)
 
         return [
             StrField(User, "username"),
@@ -447,8 +447,8 @@ class UserViewSet(UsedByMixin, ModelViewSet):
             StrField(User, "email"),
             StrField(User, "path"),
             BoolField(User, "is_active", nullable=True),
-            ChoiceSearchField(User, "type"),
-            JSONSearchField(User, "attributes", suggest_nested=False),
+            #ChoiceSearchField(User, "type"),
+            #JSONSearchField(User, "attributes", suggest_nested=False),
         ]
 
     def get_queryset(self):
diff --git a/authentik/events/api/events.py b/authentik/events/api/events.py
index 5508afa14..dc0713997 100644
--- a/authentik/events/api/events.py
+++ b/authentik/events/api/events.py
@@ -135,16 +135,16 @@ class EventViewSet(ModelViewSet):
     def get_ql_fields(self):
         from djangoql.schema import DateTimeField, StrField
 
-        from authentik.enterprise.search.fields import ChoiceSearchField, JSONSearchField
+        #from authentik.enterprise.search.fields import ChoiceSearchField, JSONSearchField
 
         return [
-            ChoiceSearchField(Event, "action"),
+            #ChoiceSearchField(Event, "action"),
             StrField(Event, "event_uuid"),
             StrField(Event, "app", suggest_options=True),
             StrField(Event, "client_ip"),
-            JSONSearchField(Event, "user", suggest_nested=False),
-            JSONSearchField(Event, "brand", suggest_nested=False),
-            JSONSearchField(Event, "context", suggest_nested=False),
+            #JSONSearchField(Event, "user", suggest_nested=False),
+            #JSONSearchField(Event, "brand", suggest_nested=False),
+            #JSONSearchField(Event, "context", suggest_nested=False),
             DateTimeField(Event, "created", suggest_options=True),
         ]
 
diff --git a/authentik/outposts/api/outposts.py b/authentik/outposts/api/outposts.py
index 318c3e3ff..cf712fb91 100644
--- a/authentik/outposts/api/outposts.py
+++ b/authentik/outposts/api/outposts.py
@@ -18,7 +18,7 @@ from authentik.core.api.providers import ProviderSerializer
 from authentik.core.api.used_by import UsedByMixin
 from authentik.core.api.utils import JSONDictField, ModelSerializer, PassiveSerializer
 from authentik.core.models import Provider
-from authentik.enterprise.license import LicenseKey
+#from authentik.enterprise.license import LicenseKey
 from authentik.lib.utils.time import timedelta_from_string, timedelta_string_validator
 from authentik.outposts.api.service_connections import ServiceConnectionSerializer
 from authentik.outposts.apps import MANAGED_OUTPOST, MANAGED_OUTPOST_NAME
@@ -139,9 +139,9 @@ class OutpostHealthSerializer(PassiveSerializer):
 
     def get_fips_enabled(self, obj: dict) -> bool | None:
         """Get FIPS enabled"""
-        if not LicenseKey.get_total().status().is_valid:
-            return None
-        return obj["fips_enabled"]
+        #if not LicenseKey.get_total().status().is_valid:
+        return None
+        #return obj["fips_enabled"]
 
 
 class OutpostFilter(FilterSet):
diff --git a/authentik/providers/scim/models.py b/authentik/providers/scim/models.py
index be5abef14..d775f84db 100644
--- a/authentik/providers/scim/models.py
+++ b/authentik/providers/scim/models.py
@@ -129,13 +129,13 @@ class SCIMProvider(OutgoingSyncProvider, BackchannelProvider):
     )
 
     def scim_auth(self) -> AuthBase:
-        if self.auth_mode == SCIMAuthenticationMode.OAUTH:
-            try:
-                from authentik.enterprise.providers.scim.auth_oauth2 import SCIMOAuthAuth
-
-                return SCIMOAuthAuth(self)
-            except ImportError:
-                LOGGER.warning("Failed to import SCIM OAuth Client")
+        #if self.auth_mode == SCIMAuthenticationMode.OAUTH:
+        #    try:
+        #        from authentik.enterprise.providers.scim.auth_oauth2 import SCIMOAuthAuth
+        #
+        #        return SCIMOAuthAuth(self)
+        #    except ImportError:
+        #        LOGGER.warning("Failed to import SCIM OAuth Client")
         return SCIMTokenAuth(self)
 
     @property
